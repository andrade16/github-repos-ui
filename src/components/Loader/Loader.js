import React from "react";
import "./Loader.scss";

const Loader = ({ message }) => {
    return (
        <div className="loader-container">
            <div className="loader" />
            <span className="loading-text">{message}</span>
        </div>
    );
};

export default Loader;
