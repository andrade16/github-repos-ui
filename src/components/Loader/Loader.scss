@import "src/utils/variables";

.loader-container {
  display: $flex-display;
  flex-direction: $flex-column;
  align-items: $content-center;

  .loader {
    margin-top: 16px;
    border-top: 10px solid $blue;
    border-right: $loader-borders;
    border-bottom: $loader-borders;
    border-left: $loader-borders;
    animation: spinner 0.9s linear infinite;
    border-radius: $large-border-radius;
    width: 64px;
    height: 64px;
  }

  .loading-text {
    margin-top: 16px;
    margin-left: 10px;
    font-style: normal;
    font-weight: normal;
    font-size: 16px;
    line-height: 130%;
    color: #888;
  }
}

@keyframes spinner {
  0% {
    transform: rotate(0deg);
  }
  100% {
    transform: rotate(360deg);
  }
}
