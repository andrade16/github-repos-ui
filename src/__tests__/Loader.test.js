import React from "react";
import renderer from "react-test-renderer";
import Loader from "../components/Loader/Loader";
import Enzyme, { shallow } from "enzyme";
import Adapter from "enzyme-adapter-react-16";

Enzyme.configure({ adapter: new Adapter() });

describe("<Loader/>", () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<Loader message="Loading..." />);
  });

  it("Renders <Loader/> correctly", () => {
    const tree = renderer.create(<Loader message="Loading..." />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders div DOM elements", () => {
    expect(wrapper.find("div")).toHaveLength(2);
  });

  it("renders span DOM elements", () => {
    expect(wrapper.find("span")).toHaveLength(1);
  });
});
