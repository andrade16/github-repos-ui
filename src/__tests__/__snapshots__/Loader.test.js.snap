// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Loader/> Renders <Loader/> correctly 1`] = `
<div
  className="loader-container"
>
  <div
    className="loader"
  />
  <span
    className="loading-text"
  >
    Loading...
  </span>
</div>
`;
